src/main/scala/viper/silver/reporter/package.scala

```diff
@@ -16,7 +18,56 @@ package object reporter {
 
   type Position = viper.silver.ast.SourcePosition
 
+
+  trait SymbolType {
+    val name: String
+  }
+
+  trait TypedSymbol extends SymbolType {
+    val viperType: Type
+  }
+
   // The following case classes are essentially named tuple wrappers.
-  case class Definition(name: String, typ: String, location: viper.silver.ast.Position,
+  case object ViperDomain extends SymbolType {
+    val name = "Domain"
+  }
+
+  case object ViperAxiom extends SymbolType {
+    val name = "Axiom"
+  }
+
+  case object ViperFunction extends SymbolType {
+    val name = "Function"
+  }
+
+  case object ViperPredicate extends SymbolType {
+    val name = "Predicate"
+  }
+
+  case class ViperField(viperType: Type) extends TypedSymbol {
+    val name = "Field"
+  }
+
+  case object ViperMethod extends SymbolType {
+    val name = "Method"
+  }
+
+  case class ViperArgument(viperType: Type) extends TypedSymbol {
+    val name = "Argument"
+  }
+
+  case class ViperReturnParameter(viperType: Type) extends TypedSymbol {
+    val name = "Return"
+  }
+
+  case object ViperUntypedLocalDefinition extends SymbolType {
+    val name = "Local"
+  }
+
+  case class ViperTypedLocalDefinition(viperType: Type) extends TypedSymbol {
+    val name = "Local"
+  }
+
+  case class Definition(name: String, typ: SymbolType, location: viper.silver.ast.Position,
                         scope: Option[viper.silver.ast.AbstractSourcePosition] = None)
```

Comment:
Wouldn't it make more sense to type `scope` as `Option[viper.silver.ast.Scope]`?

Reply:
Nice catch. I bet the original code had been written before we introduced proper Viper scopes.

Reply:
On second thoughts, the Scope field in the Reporter package is basically just a pair of ```AbstractSourcePosition``` which can be easily serialized. The ```viper.silver.ast.Scope``` carries a ```Seq``` of ```scopedDecls``` which is not what we need here.